cmd/otel-allocator/allocation/per_node.go

```diff
@@ -190,12 +190,8 @@ func (allocator *perNodeAllocator) handleTargets(diff diff.Changes[*target.Item]
 
 	// Check for unassigned targets
 	if len(unassignedTargetsForJobs) > 0 {
```

Comment:
this can just become an integer that's incremented on line 186, i don't think there's a need for a map, unless we care about deduping (which should happen already in the filter strategy)

Reply:
I was wondering if there can be a scenario where we have multiple targets per job, that cannot be assigned. But I see the point, I simplified this to just count the number of targets, that should hopefully give enough info when debugging.